bika/lims/content/analysisrequest.py

```diff
@@ -2473,7 +2473,7 @@ def getProgress(self):
         """Returns the progress in percent of all analyses
         """
         review_state = api.get_review_status(self)
-        if review_state == "published":
+        if review_state in ["published", "invalid"]:
```

Comment:
Shall we consider the `cancelled` state as well as complete?

Reply:
I wouldn't. "invalid" status can only be achieved after sample gets verified or published (so all analyses have been completed). "cancelled" happens in earlier stages and I think is ok to let the user now the percentage of conpleteness when the sample was cancelled, rather than forcing 100%

Reply:
Please consider that progresses less than 100% indicate that some actions still have to be done, which is not the case for the cancelled samples.
In our case we cancel samples, when there is no more interest in them, so I would expect a 100% when all active samples are done.
I'm not sure what to do best with the "rejected" ones...

Reply:
I would agree with @grulisco to interpret all final states as 100% done. Reinstantiate e.g. cancelled Samples would anyhow correctly restore the progress to the actual state of the Analyses.

Reply:
If the criteria is that the progress indicates that no more actions are required (instead of the percentage of completed actions), then we should always apply a percentage of 100% to end-statuses, these being  `rejected`, `cancelled` or `published`. 

IMO, percentage of completeness fits better with the number of tasks/actions done over the number of expected tasks/actions: for me a 100% completed for a `rejected` or `cancelled` sample is not intuitive, but well, a matter of taste I guess. I'd say this confusion can only happen when you are listing Samples with filter "All" enabled (you'll never see `rejected`/`cancelled` samples when "Active"/"To be verified"/... filter is used).

Is ok to me if you feel is better the percentage to indicate if there are still actions to be done, but then I strongly suggest to apply same criteria for `rejected`.

Reply:
> I'd say this confusion can only happen when you are listing Samples with filter "All" enabled (you'll never see `rejected`/`cancelled` samples when "Active"/"To be verified"/... filter is used).

Full ACK in samples listing, but we are listing batches here. So confusion will definitely happen.
This feature is useful as an overview on **open** batches only, because closed ones are always to be expected 100%.

I also agree with 100% for all end-status...

Reply:
The progress calculation in batches relies on the code from Sample, which means that both will behave consistently in the end. I'm going then to consider all final sample states as 100%, this will then make a batch 100% even when it contains cancelled/rejected/retracted samples next to some published ones.